### PR TITLE
Reverts pin to string to accomodate pins starting with 0

### PR DESCRIPTION
**Pull request by akolson**
## Summary

This pull request reverts pin to be stored as string due to the failure to store an integer pin `0000` as such values would be truncated of all zeros and the most significant value retained. This fails the length validation.

## References

Contributes to #9789 

## Reviewer guidance

Please see https://github.com/learningequality/kolibri/pull/10031

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [x] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/auth/test/test_api.py [hunk 1]
```diff
@@ -1501,24 +1507,24 @@ def test_facility_admin_can_check_is_pin_valid_unset_pin(self):
...
 
-        response = self.is_pin_valid({"pin_code": 1234})
+        response = self.is_pin_valid({"pin_code": "1234"})
         self.assertEqual(response.data["is_pin_valid"], False)
 
     def test_facility_admin_can_check_is_pin_valid_empty_pin_specified(self):
         self.client.login(username=self.superuser.username, password=DUMMY_PASSWORD)
-        response = self.update_pin({"pin_code": 1234})
-        response = self.is_pin_valid({"pin_code": None})
+        response = self.update_pin({"pin_code": "1234"})
```
Comment (nucleogenesis) [inline kolibri/core/auth/test/test_api.py]: There are several spots in these tests where you assign `response = self.update_pin(...)` but then never end up using _that_ `response` variable but instead overwrite it shortly after. Just calling `self.update_pin` directly should be fine in any case where you're not asserting against that particular response.

Super minor and not a blocker.

**Review by nucleogenesis — APPROVED** [review]
Looks good! :+1:

File: kolibri/core/auth/test/test_api.py (hunk 1 above)
Comment (akolson) [reply kolibri/core/auth/test/test_api.py]: Thanks @nucleogenesis  for pointing this out. I have done cleanup on these as they are minor changes

**Review by bjester — APPROVED** [review]
Sorry about the regression, looks good

**Pull request merged**